### PR TITLE
MLP forward backward

### PR DESCRIPTION
**Pull request by panyx0718**
test=develop

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -9426,3 +9427,47 @@ def huber_loss(input, label, delta):
...
+    def __init__(self,
+                 size,
+                 param_attr=None,
+                 num_flatten_dims=1,
+                 dtype=core.VarDesc.VarType.FP32):
+        super(FC, self).__init__()
+        self._size = size
+        self._num_flatten_dims = num_flatten_dims
+        self._dtype = dtype
+        self._helper = LayerHelper('FC', param_attr=param_attr)
```
Comment (velconia) [inline python/paddle/fluid/layers/nn.py:9442]: NO need to export layer helper to user, Better put to PyLayer

File: paddle/fluid/imperative/tracer.h [hunk 2]
```diff
@@ -43,9 +43,12 @@ void CreateGradOp(const framework::OpDesc& op_desc,
 
 class Tracer {
  public:
-  explicit Tracer(framework::BlockDesc* root_block) : root_block_(root_block) {
+  explicit Tracer(framework::BlockDesc* root_block,
+                  framework::BlockDesc* startup_block)
+      : root_block_(root_block), startup_block_(startup_block) {
     root_scope_ = new framework::Scope();
     scopes_[root_block_] = root_scope_;
+    scopes_[startup_block_] = root_scope_;
```
Comment (velconia) [inline paddle/fluid/imperative/tracer.h:51]: Different opinion here, two different blocks ```root_block``` and ```startup_block``` will involve complexity and NO benefits, Parameter initiliazation is better to put into ```_build_once```

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -9426,3 +9427,47 @@ def huber_loss(input, label, delta):
                  'Residual': residual},
         attrs={'delta': delta})
     return out
+
+
+class FC(layers.PyLayer):
```
Comment (velconia) [inline python/paddle/fluid/layers/nn.py:9432]: Better to put this layer into ```imperative.nn``` or something

**Review by velconia — APPROVED** [review]
LGTM with a little comments

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (panyx0718) [reply python/paddle/fluid/layers/nn.py:9432]: Currently, we are not exposing it via __all__. but eventually we want to make PyLayer support both static and dynamic

**Review by velconia — COMMENTED** [review]
LGTM, polish the code later

**Pull request merged**